### PR TITLE
Add rcodesign installation to golang-builder

### PR DESCRIPTION
**Pull request by gitperr**
Preparation for code signing OSX builds, will help fixing issues like https://github.com/prometheus/node_exporter/issues/2539.

See https://github.com/indygreg/apple-platform-rs/tree/main/apple-codesign#rcodesign-cli for more info.

### CODE REVIEW

**Comment by gitperr** [comment]
@SuperQ did you have any feedback for this?

File: 1.20/main/Dockerfile [hunk 1]
```diff
@@ -22,10 +22,12 @@ RUN \
...
         crossbuild-essential-ppc64el linux-libc-dev-ppc64el-cross \
         crossbuild-essential-s390x linux-libc-dev-s390x-cross \
-        gcc-riscv64-linux-gnu g++-riscv64-linux-gnu libc6-dev-riscv64-cross\
-        libc6-riscv64-cross linux-libc-dev-riscv64-cross\
+        gcc-riscv64-linux-gnu g++-riscv64-linux-gnu libc6-dev-riscv64-cross \
+        libc6-riscv64-cross linux-libc-dev-riscv64-cross \
     && rm -rf /var/lib/apt/lists/* \
-    && mkdir -p /tmp/osxcross
+    && mkdir -p /tmp/osxcross \
+    && curl https://sh.rustup.rs -sSf | sh -s -- -y
```
Comment (SuperQ) [inline 1.20/main/Dockerfile:29]: I would rather use a pre-built binary, or add a build image step like:

```
FROM rust:latest as rustbuilder

RUN cargo...

FROM quay.io/prometheus/golang-builder:1.20-base

COPY --from=rustbuilder ...

```

File: 1.20/main/Dockerfile [hunk 2]
```diff
@@ -22,10 +26,13 @@ RUN \
...
-        gcc-riscv64-linux-gnu g++-riscv64-linux-gnu libc6-dev-riscv64-cross\
-        libc6-riscv64-cross linux-libc-dev-riscv64-cross\
+        gcc-riscv64-linux-gnu g++-riscv64-linux-gnu libc6-dev-riscv64-cross \
+        libc6-riscv64-cross linux-libc-dev-riscv64-cross \
     && rm -rf /var/lib/apt/lists/* \
-    && mkdir -p /tmp/osxcross
+    && mkdir -p /tmp/osxcross \
+    && mkdir -p /usr/local/cargo/bin
+    
+COPY --from=rustbuilder /usr/local/cargo/bin/rcodesign /usr/local/cargo/bin/rcodesign
```
Comment (SuperQ) [inline 1.20/main/Dockerfile:35]: Why not just copy this to `/usr/local/bin`?

File: 1.21/main/Dockerfile [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
+RUN cargo install --git https://github.com/indygreg/apple-platform-rs --branch main --bin rcodesign apple-codesign
+
 FROM        quay.io/prometheus/golang-builder:1.21-base
 MAINTAINER  The Prometheus Authors <prometheus-developers@googlegroups.com>
 
 RUN \
     apt-get update && apt-get install -y --no-install-recommends \
         clang \
-        cmake \
+        cmake \
```
Comment (SuperQ) [inline 1.21/main/Dockerfile:11]:
```suggestion
        cmake \
```

**Review by SuperQ — APPROVED** [review]
Minor nit, otherwise LGTM.

File: 1.21/main/Dockerfile (hunk 3 above)
Comment (gitperr) [reply 1.21/main/Dockerfile]: Got rid of that :+1:

**Pull request merged**